pandas/core/frame.py

```diff
@@ -7390,7 +7390,9 @@ def f(x):
             return op(x, axis=axis, skipna=skipna, **kwds)
 
         # exclude timedelta/datetime unless we are uniform types
-        if axis == 1 and self._is_mixed_type and self._is_datelike_mixed_type:
+        if (axis == 1 and self._is_datelike_mixed_type
+                and (not self._is_homogeneous_type
+                     and not self._data.blocks[0].is_datetimetz)):
```

Comment:
really don't like the last condition as this is reaching into internals. are the prior ones not sufficient?

Reply:
Alternative is `is_datetimetz64_dtype(self.dtypes[0])`. I don't have a preference on which is cleaner.

Reply:
Opted for your suggestion @TomAugspurger. Looks a little cleaner from first inspection.